SummerSchool/app/src/main/java/nl/rug/www/rugsummerschools/controller/MainActivity.java

```diff
@@ -0,0 +1,386 @@
+import com.facebook.AccessToken;
+import com.facebook.CallbackManager;
+import com.facebook.FacebookCallback;
+import com.facebook.FacebookException;
+import com.facebook.login.LoginManager;
+import com.facebook.login.LoginResult;
+import com.facebook.login.widget.LoginButton;
+import com.google.android.gms.auth.api.signin.GoogleSignIn;
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+import com.google.android.gms.auth.api.signin.GoogleSignInClient;
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
+import com.google.android.gms.common.api.ApiException;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthCredential;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FacebookAuthProvider;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.auth.GoogleAuthProvider;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+
+import nl.rug.www.rugsummerschools.R;
+import nl.rug.www.rugsummerschools.controller.announcement.AnnouncementListFragment;
+import nl.rug.www.rugsummerschools.controller.forum.ForumLoginFragment;
+import nl.rug.www.rugsummerschools.controller.forum.ForumThreadListFragment;
+import nl.rug.www.rugsummerschools.controller.generalinfo.GeneralInfoListFragment;
+import nl.rug.www.rugsummerschools.controller.lecturer.LecturerListFragment;
+import nl.rug.www.rugsummerschools.controller.timetable.TimeTableFragment;
+
+/**
+ * This class is main activity that contains basic layout of the app.
+ * It consists of header, viewpager, and buttons.
+ * Fragments (Announcements, Generalninfos, ...) will be inflated in view pager.
+ *
+ * @since 13/04/2017
+ * @author Jeongkyun Oh
+ */
+
+public class MainActivity extends AppCompatActivity implements ForumLoginFragment.OnSignInListener, ForumThreadListFragment.OnSignOutListener {
+
+    private static final String TAG = "MainActivity";
+    private static final int FRAGMENTS_SIZE = 5;
+
+    private static final int RC_SIGN_IN = 9001;
+
+    private FirebaseAuth mAuth;
+    private LoginButton mInvisibleFacebookLoginButton;
+    private CallbackManager mCallbackManager;
+    private GoogleSignInClient mGoogleSignInClient;
+    private ViewPager mViewPager;
+    private BottomNavigationView mNavigation;
+
+
+    private Fragment[] mFragments = {
+            new AnnouncementListFragment(),
+            new GeneralInfoListFragment(),
+            new LecturerListFragment(),
+            new TimeTableFragment(),
+            new ForumLoginFragment(),
+            new ForumThreadListFragment()
+    };
+
+    private FragmentStatePagerAdapter mFragmentStatePagerAdapter = new FragmentStatePagerAdapter(getSupportFragmentManager()) {
+        @Override
+        public Fragment getItem(int position) {
+            switch (position) {
+                case 0 : return mFragments[0];
```

Comment:
Wouldn't it make sense here to use CONSTANT_SYMBOLIC_NAMES instead of integers?

Reply:
Right, I agree with that. I will change it.